plugins/SegmentEditor/SegmentEditor.php

```diff
@@ -71,6 +85,92 @@ public function getKnownSegmentsToArchiveForSite(&$segments, $idSite)
         $segments = array_unique($segments);
     }
 
+    public function onNoArchiveData()
+    {
+        $segmentInfo = $this->getSegmentIfIsUnprocessed();
+        if (empty($segmentInfo)) {
+            return;
+        }
+
+        list($segment, $isSegmentToPreprocess) = $segmentInfo;
+
+        // this archive has no data, the report is for a segment that gets preprocessed, and the archive for this
+        // data does not exist. this means the data will be processed later. we let the user know so they will not
+        // be confused.
+        $model = new Model();
+        $storedSegment = $model->getSegmentByDefinition($segment->getString()) ?: null;
+
+        throw new UnprocessedSegmentException($segment, $isSegmentToPreprocess, $storedSegment);
+    }
+
+    public function onLoadingError(\Exception $ex, View $dataTableView)
+    {
+        if (!($ex instanceof UnprocessedSegmentException)) {
+            Notification\Manager::cancel(self::NO_DATA_UNPROCESSED_SEGMENT_ID);
+            return;
+        }
+
+        $segment = $ex->getSegment();
+        $segmentToPreprocess = $ex->getStoredSegment();
+
+        $segmentDisplayName = !empty($segmentToPreprocess['name']) ? $segmentToPreprocess['name'] : $segment->getString();
+
+        $view = new View('@SegmentEditor/_unprocessedSegmentMessage.twig');
+        $view->isSegmentToPreprocess = $ex->isSegmentToPreprocess();
+        $view->segmentName = $segmentDisplayName;
+        $view->visitorLogLink = '#' . Url::getCurrentQueryStringWithParametersModified([
+            'category' => 'General_Visitors',
+            'subcategory' => 'Live_VisitorLog',
+        ]);
+
+        $notification = new Notification($view->render());
+        $notification->priority = Notification::PRIORITY_HIGH;
+        $notification->context = Notification::CONTEXT_INFO;
+        $notification->flags = Notification::FLAG_NO_CLEAR;
+        $notification->type = Notification::TYPE_TRANSIENT;
+        $notification->raw = true;
+
+        unset($dataTableView->error); // don't display the API error
+        $dataTableView->notifications[self::NO_DATA_UNPROCESSED_SEGMENT_ID] = $notification;
+    }
+
+    private function getSegmentIfIsUnprocessed()
```

Comment:
Maybe this method could be made reusable similar to `Segment::willBeArchived` and / or maybe the logic can or has to be merged?

Reply:
Do you see potential for reuse w/ this logic? I'm not sure where it would go if made re-usable.